### ISSUE TITLE
No ciphers enabled for max supported SSL/TLS version (even after disabling TLS 1.3 or calling SSL_CTX_set_ciphersuites)

### ISSUE THREAD

**Issue by okbaz** [openssl/openssl#8290]
I am trying to choose the cipher suites to be used for a client implementation.
I set the min TLS version to 1.1 and the max TLS version to 1.2
    ```
    SSL_CTX_set_min_proto_version(ctx,TLS1_1_VERSION);
    SSL_CTX_set_max_proto_version(ctx,TLS1_2_VERSION);
    ```

I set the cipher suite as the following:

```
   if ( SSL_CTX_set_cipher_list(ctx,"TLS_RSA_WITH_AES_128_CBC_SHA") == 1)
        printf("OK\n");
```

Unfortunately I am still getting the erorr **"No ciphers enabled for max supported SSL/TLS version"** 

I even set the TLS 1.3 cipher suites explicitly as following and also I still have the same error

`SSL_CTX_set_ciphersuites(ctx,"TLS_CHACHA20_POLY1305_SHA256");`

To conclude I configured the client to use TLS 1.1 and 1.2 and set explicitly the cipher suites to be used, and I even set TLS 1.3 cipher suites explicitly and still have **"No ciphers enabled for max supported SSL/TLS version"**, How could I solve this problem?

**Comment by mattcaswell** [comment]
`TLS_RSA_WITH_AES_128_CBC_SHA` is not a valid ciphersuite name in the OpenSSL world. For historical reasons OpenSSL uses its own names for many ciphersuites (not standard RFC names). See the following man page for the mapping from RFC name to OpenSSL name:

https://www.openssl.org/docs/man1.1.1/man1/ciphers.html

In this case the ciphersuite you want is called `AES128-SHA`.

Unfortunately, due to a bug in 1.1.1a and below, the `SSL_CTX_set_cipher_list()` call still returns a success response even if it doesn't match any ciphersuites. This is fixed in the forthcoming 1.1.1b (due to be released next week).

So what is happening here is that your `SSL_CTX_set_cipher_list()` call is reporting success but setting no ciphersuites for TLSv1.2/1.1 hence the subsequent "No ciphers enabled for max supported SSL/TLS version" error.

**Comment by mattcaswell** [comment]
This question seems to have been answered. Closing.
